energy-dao/src/external_sc_interactions/farm_actions.rs

```diff
@@ -0,0 +1,161 @@
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+{
+    fn call_enter_farm(
+        &self,
+        farm_address: ManagedAddress,
+        farming_tokens: PaymentsVec<Self::Api>,
+    ) -> EnterFarmResultType<Self::Api> {
+        self.farm_proxy(farm_address)
+            .enter_farm_endpoint(OptionalValue::<ManagedAddress>::None)
+            .with_multi_token_transfer(farming_tokens)
+            .execute_on_dest_context()
+    }
+
+    fn call_exit_farm(
+        &self,
+        farm_address: ManagedAddress,
+        farm_tokens: EsdtTokenPayment,
+    ) -> ExitFarmWithPartialPosResultType<Self::Api> {
+        self.farm_proxy(farm_address)
+            .exit_farm_endpoint(
+                farm_tokens.amount.clone(),
+                OptionalValue::<ManagedAddress>::None,
+            )
+            .with_esdt_transfer(farm_tokens)
+            .execute_on_dest_context()
+    }
+
+    fn call_farm_claim(
+        &self,
+        farm_addr: ManagedAddress,
+        farm_token: EsdtTokenPayment,
+    ) -> ClaimRewardsResultType<Self::Api> {
+        self.farm_proxy(farm_addr)
+            .claim_rewards_endpoint(OptionalValue::<ManagedAddress>::None)
+            .with_esdt_transfer(farm_token)
+            .execute_on_dest_context()
+    }
+
+    fn update_farm_after_claim(
+        &self,
+        initial_farm_state: &FarmState<Self::Api>,
+        farm_state_mapper: &mut SingleValueMapper<FarmState<Self::Api>>,
+        new_farm_token: EsdtTokenPayment,
+        farm_rewards: EsdtTokenPayment,
+        division_safety_constant: &BigUint,
+    ) {
+        farm_state_mapper.update(|config| {
+            if farm_rewards.amount > 0 {
+                let rps_increase = self.compute_farm_rps_increase(
+                    &farm_rewards.amount,
+                    &new_farm_token.amount,
+                    division_safety_constant,
+                );
+
+                let new_rewards = if initial_farm_state.reward_reserve > 0 {
+                    let mut reward_payments = ManagedVec::new();
+                    let current_rewards = EsdtTokenPayment::new(
+                        farm_rewards.token_identifier.clone(),
+                        initial_farm_state.reward_token_nonce,
+                        initial_farm_state.reward_reserve.clone(),
+                    );
+                    reward_payments.push(farm_rewards);
+                    reward_payments.push(current_rewards);
+                    self.merge_locked_tokens(reward_payments)
+                } else {
+                    farm_rewards
+                };
+
+                config.reward_token_nonce = new_rewards.token_nonce;
+                config.reward_reserve = new_rewards.amount;
+                config.farm_rps += rps_increase;
+            }
+
+            config.farm_staked_value = new_farm_token.amount;
+            config.farm_token_nonce = new_farm_token.token_nonce;
+        });
```

Comment:
Reduce nesting:

```suggestion
            config.farm_staked_value = new_farm_token.amount;
            config.farm_token_nonce = new_farm_token.token_nonce;
            
            if farm_rewards.amount > 0 {
                     return;
            }

            
                let rps_increase = self.compute_farm_rps_increase(
                    &farm_rewards.amount,
                    &new_farm_token.amount,
                    division_safety_constant,
                );

                let new_rewards = if initial_farm_state.reward_reserve > 0 {
                    let mut reward_payments = ManagedVec::new();
                    let current_rewards = EsdtTokenPayment::new(
                        farm_rewards.token_identifier.clone(),
                        initial_farm_state.reward_token_nonce,
                        initial_farm_state.reward_reserve.clone(),
                    );
                    reward_payments.push(farm_rewards);
                    reward_payments.push(current_rewards);
                    self.merge_locked_tokens(reward_payments)
                } else {
                    farm_rewards
                };

                config.reward_token_nonce = new_rewards.token_nonce;
                config.reward_reserve = new_rewards.amount;
                config.farm_rps += rps_increase;
        });
```

Reply:
Github code edit sucks, so also please format the code afterwards.

Reply:
Updated, but slightly different, as I need to set `farm_state_mapper` in both cases.